scripts/forecast_csv.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pandas as pd
+from datetime import datetime, timedelta
+from quartz_solar_forecast.forecast import run_forecast
+from quartz_solar_forecast.pydantic_models import PVSite
+import unittest
+from unittest.mock import patch
+from datetime import datetime
+
+def generate_forecast(init_time_freq, start_datetime, end_datetime, site_name):
+    """
+    Generates forecasts at specified intervals and saves them into a CSV file.
+
+    Args:
+        init_time_freq (int): The frequency in hours at which the forecasts are generated.
+        start_datetime (str): The starting date and time for generating forecasts.
+        end_datetime (str): The ending date and time for generating forecasts.
+        site_name (str): The name of the site for which the forecasts are generated.
+    """
+    start = datetime.strptime(start_datetime, "%Y-%m-%d %H:%M:%S")
+    start_date = start.date()
+    end = datetime.strptime(end_datetime, "%Y-%m-%d %H:%M:%S")
+    end_date = end.date()
+    all_forecasts = pd.DataFrame()
+    site = PVSite(latitude=51.75, longitude=-1.25, capacity_kwp=1.25)
```

Comment:
do you think lat and lon capacity, could be inputs to the function? This way it is even more generic?

Reply:
> do you think lat and lon capacity, could be inputs to the function? This way it is even more generic?

That's a really good idea!

Reply:
@peterdudfield I think we can merge it into the main library at a later date. Good idea to move the lat, lon and capacity to be inputs for the function.